### ISSUE TITLE
Handling Markdown tables in Julia 1.0

### ISSUE THREAD

**Issue by dmbates** [mpastell/Weave.jl#149]
This looks like it might have a straightforward fix.  I have a Markdown table in my `.jmd` source and called
```julia
julia> weave("MRK_Exp1.jmd", doctype="md2html")
```

The weave part went fine but I got an error of
```
ERROR: MethodError: no method matching show(::Base.GenericIOBuffer{Array{UInt8,1}}, ::MIME{Symbol("text/plain")}, ::Markdown.Table)
You may have intended to import Base.show
Closest candidates are:
  show(::IO, ::MIME{Symbol("text/html")}, ::Markdown.MD) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:230
Stacktrace:
 [1] #sprint#325(::Nothing, ::Int64, ::Function, ::Function, ::MIME{Symbol("text/plain")}, ::Vararg{Any,N} where N) at ./strings/io.jl:101
 [2] sprint(::Function, ::MIME{Symbol("text/plain")}, ::Vararg{Any,N} where N) at ./strings/io.jl:97
 [3] tohtml(::Base.GenericIOBuffer{Array{UInt8,1}}, ::MIME{Symbol("text/plain")}, ::Markdown.Table) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:12
 [4] tohtml(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Markdown.Table) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:34
 [5] html(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Markdown.Table) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:164
 [6] html(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Array{Any,1}) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:87
 [7] html(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Markdown.MD) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:92
 [8] #sprint#325(::Nothing, ::Int64, ::Function, ::Function, ::Markdown.MD) at ./strings/io.jl:101
 [9] sprint at ./strings/io.jl:97 [inlined]
 [10] html(::Markdown.MD) at /home/bates/.julia/packages/Weave/1O5IK/src/Markdown2HTML.jl:227
 [11] format_chunk(::Weave.DocChunk, ::Dict{Symbol,Any}, ::Weave.JMarkdown2HTML) at /home/bates/.julia/packages/Weave/1O5IK/src/format.jl:132
 [12] format(::Weave.WeaveDoc) at /home/bates/.julia/packages/Weave/1O5IK/src/format.jl:33
 [13] #weave#10(::String, ::Symbol, ::Symbol, ::Dict{Any,Any}, ::Symbol, ::String, ::Nothing, ::String, ::Symbol, ::Bool, ::Nothing, ::Nothing, ::Nothing, ::Array{String,1}, ::String, ::typeof(weave), ::String) at /home/bates/.julia/packages/Weave/1O5IK/src/Weave.jl:112
 [14] (::getfield(Weave, Symbol("#kw##weave")))(::NamedTuple{(:doctype,),Tuple{String}}, ::typeof(weave), ::String) at ./none:0
 [15] top-level scope at none:0
```
during the conversion to Markdown.  I'll see if I can create a PR.

**Comment by dmbates** [comment]
I was able to get past the error by changing `show` to `Base.show` in the line cited at [3] of the stack trace but that doesn't fix the underlying problem.  It appears that it should be the `tohtml` method with `m::MIME"text/html"` that is called, not the one with `m::MIME"text/plain"`.  It looks like `bestmime` is returning `MIME"text/plain"`, not `MIME"text/html"` for `Markdown.Table` so I think I need to trace where `showable` is defined.

This is turning into a rabbit hole for me.  I will need to look at it later unless someone else with more knowledge of the package can track things down first.

**Comment by mpastell** [comment]
This should be fixed in #165 show I'll close this issue. Feel to reopen if the problem persists!
